### ISSUE TITLE
Fails in this case

### ISSUE THREAD

**Issue by JoePerkins** [Layman806/MST-Prim#1]
Wrong result with this input:

> 9 14
> 1 2 4
> 1 8 9
> 2 3 9
> 2 8 11
> 3 4 7
> 3 9 2
> 3 6 4
> 4 5 10
> 4 6 15
> 5 6 11
> 6 7 2
> 7 8 1
> 7 9 6
> 8 9 7

Result:

> ::MST::
> Root is :1
> ( 2, 1 ) : 4
> ( 3, 9 ) : 2
> ( 4, 3 ) : 7
> ( 5, 4 ) : 10
> ( 6, 7 ) : 2
> ( 7, 8 ) : 1
> ( 8, 7 ) : 1
> ( 9, 3 ) : 2

**Comment by sagar2009kumar** [comment]
Yes, I think that it must fails Because it is not updating the given edge whenever the edge is reduced. The better approach would be: 

Create a set of completing vertex and always insert the weight into the queue (Whatever happens)  if the  node is already completed then just remove the edge and do nothing else proceed as in MST. 

Hope it helps.

**Comment by Layman806** [comment]
I will update this code once I regain access to this original account. I messed up the 2 Factor Authentication in this account, so using another account temporarily.
